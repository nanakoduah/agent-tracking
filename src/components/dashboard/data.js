const kpiData = [
  { target: 2000, achieved: 88, name: 'payment' },
  { target: 1500, achieved: 58, name: 'customers' },
  { target: 20, achieved: 12, name: 'revenue' },
  { target: 1000, achieved: 76, name: 'saving' }
];

const agentsData = [
  {
    agentCode: '7XTWEGDGTYWTE',
    agentName: 'BABA FUSEINI',
    location: { lat: 8.0, lng: -2.3 }
  },
  {
    agentCode: 'HFSUYEWWEY89',
    agentName: 'KOFI KABOKO',
    location: { lat: 7.8, lng: -2.1594 }
  },
  {
    agentCode: '63RRYRTFGH',
    agentName: 'BABA FUSEINI',
    location: { lat: 7.98, lng: -1.99 }
  },
  {
    agentCode: 'RTUSDGD',
    agentName: 'DUGS MYKEL',
    location: { lat: 7.5, lng: -2.0094 }
  }
];

export { kpiData, agentsData };
