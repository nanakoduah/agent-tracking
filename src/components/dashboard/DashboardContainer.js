import React from 'react';
import PropTypes from 'prop-types';

import { kpiData, agentsData } from '../../__mocks__/data';
import { KPIShape, agentShape } from '../common-props';

class DashboardContainer extends React.Component {
  static propTypes = {
    component: PropTypes.func.isRequired,
    kpis: PropTypes.arrayOf(KPIShape),
    agents: PropTypes.arrayOf(agentShape)
  };

  static defaultProps = {
    kpis: kpiData,
    agents: agentsData
  };

  state = {
    selectedAgents: []
  };

  handleAgentSelection = selectedAgents => this.setState({ selectedAgents });

  render() {
    const { kpis, agents, component: Component } = this.props;
    const { selectedAgents } = this.state;

    return (
      <Component
        kpis={kpis}
        agents={agents}
        selectedAgents={selectedAgents}
        onAgentSelectionChange={this.handleAgentSelection}
      />
    );
  }
}

export default DashboardContainer;
