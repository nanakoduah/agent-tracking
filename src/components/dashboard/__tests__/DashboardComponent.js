import React from 'react';
import renderer from 'react-test-renderer';

import DashboardContainer from '../DashboardContainer';
import { kpiData, agentsData } from '../../../__mocks__/data';

describe('Dashboard Container', () => {
  it('Should render properly', () => {
    const tree = renderer.create(
      <DashboardContainer
        component={jest.fn(() => null)}
        kpis={kpiData}
        agents={agentsData}
      />
    );
    expect(tree.getInstance().state).toMatchSnapshot();
  });

  it.only('Should handle agent selection properly', () => {
    const component = jest.fn(() => null);
    const tree = renderer.create(
      <DashboardContainer
        component={component}
        kpis={kpiData}
        agents={agentsData}
      />
    );

    const instance = tree.getInstance();

    const { onAgentSelectionChange } = component.mock.calls[0][0];
    onAgentSelectionChange(agentsData);

    expect(instance.state.selectedAgents).toEqual(agentsData);
  });
});
