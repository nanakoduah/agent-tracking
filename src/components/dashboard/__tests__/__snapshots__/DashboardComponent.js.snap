// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Container Should render properly 1`] = `
Object {
  "selectedAgents": Array [],
}
`;
